File: tools/license-maker.mjs
/* Typographus — license key generator (Nexflamma internal).
   Mints an offline JWT bound to a device HWID.

   Usage:
     node tools/license-maker.mjs <HWID> [days] [plan]
   Examples:
     node tools/license-maker.mjs A1B2C3D4E5F60718 365 Full
     node tools/license-maker.mjs A1B2C3D4E5F60718 0   Perpetua   (0 = nessuna scadenza)
*/

import { createRequire } from "module";
const require = createRequire(import.meta.url);
const lic = require("../electron/license.cjs");

const [hwid, daysArg, planArg] = process.argv.slice(2);

if (!hwid) {
  console.error("Uso: node tools/license-maker.mjs <HWID> [giorni] [piano]");
  process.exit(1);
}

const days = daysArg === undefined ? 365 : Number(daysArg);
const plan = planArg || "Full";
const now = Math.floor(Date.now() / 1000);
const payload = {
  hwid,
  plan,
  iss: "Nexflamma S.r.l.",
  product: "Typographus",
  iat: now,
};
if (days > 0) payload.exp = now + days * 86400;

const token = lic.signToken(payload);

console.log("\nHWID:   ", hwid);
console.log("Piano:  ", plan);
console.log("Scadenza:", days > 0 ? new Date(payload.exp * 1000).toLocaleDateString("it-IT") : "perpetua");
console.log("\n--- CHIAVE DI LICENZA ---\n");
console.log(token);
console.log("");

File: tools/license_maker.py
"""
Typographus license generator (Nexflamma, offline).

Usage:
  python tools/license_maker.py <HWID> [giorni] [piano]
    giorni = 0  -> licenza perpetua (default)
  python tools/license_maker.py            -> usa l'HWID di QUESTA macchina

Stampa la chiave. Con --save la salva anche nel percorso licenza locale.
"""
import os
import sys
import time

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from core import license_manager as lic


def main():
    args = [a for a in sys.argv[1:] if a != "--save"]
    save = "--save" in sys.argv

    hwid = args[0] if len(args) > 0 else lic.get_hwid()
    days = int(args[1]) if len(args) > 1 else 0
    plan = args[2] if len(args) > 2 else "Full"

    payload = {"hwid": hwid, "plan": plan}
    if len(args) > 3:
        payload["email"] = args[3]
    if days > 0:
        payload["exp"] = int(time.time()) + days * 86400

    token = lic.sign_token(payload)
    print(token)
    if save:
        lic.save_token(token)
        print(f"[saved] {lic.license_path()}", file=sys.stderr)


if __name__ == "__main__":
    main()
